GPU/Math3D.h

```diff
@@ -1363,6 +1377,59 @@ inline Vec3<float> Vec3<float>::operator * (const float &other) const {
 	return Vec3<float>(_mm_mul_ps(SAFE_M128(vec), _mm_set_ps1(other)));
 }
 
+// Vec4<int> operation
+template<>
+inline Vec4<int> Vec4<int>::operator + (const Vec4 &other) const {
+	return Vec4<int>(_mm_add_epi32(SAFE_M128I(ivec), SAFE_M128I(other.ivec)));
+}
+
+template<>
+inline Vec4<int> Vec4<int>::operator * (const Vec4 &other) const {
+	__m128i a = SAFE_M128I(ivec);
+	__m128i b = SAFE_M128I(other.ivec);
+#if _M_SSE >= 0x401
+	return Vec4<int>(_mm_mullo_epi32(a, b));
+#else
+	// This is what clang does. Seems about as good
+	// as it gets.
+	__m128i m02 = _mm_mul_epu32(a, b);
+	__m128i m13 = _mm_mul_epu32( // 0xF5 -> [1, 1, 3, 3]
```

Comment:
nit: Would prefer using `_MM_SHUFFLE()` even though it is "backwards".

-[Unknown]